### DISCUSSION TITLE
Add support to change the Controller / useController "ref" clean values function to accomodate more libraries

### DISCUSSION THREAD

**Discussion by jacomonh** [react-hook-form/react-hook-form#10081] [Q&A]
**Is your feature request related to a problem? Please describe.**
I'm using react-select and react-hook-form. I created a custom Select and I am managing it using a useController. However, when I use the "reset" function from useForm, it does not clean the dataon react-select. instead, I had to kinda use a hack like this:

`const watchedValues = watch();
Object.keys(watchedValues).forEach((key) => {
                    setValue(key as keyof typeof watchedValues, null);
                  });`

**Describe the solution you'd like**
Be able to manage the "clear values" function for more usability with other libraries like the life saving react-select

**Describe alternatives you've considered**
using useImperativeHandle to amange the ref from useController (not succesfull)

**Additional context**

**Comment by bluebill1049** [comment]
could you please double-check our example in the example? I am pretty much we cover that usage already.

**Comment by jacomonh** [comment]
Do you mean this example?
https://github.com/react-hook-form/react-hook-form/blob/master/examples/V7/customInput.tsx

Or the one on the website?

Bc I can't seem to figure out why the reset does not work on react-select

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/usecontroller/controller/

The following codesandbox demonstrate the usages:

[MUI and other components](https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5)

[Chakra UI components](https://codesandbox.io/s/chakra-ui-5mp8g)
